src/media.rs

```diff
@@ -33,6 +37,26 @@ impl Media {
       extensions.join(" "),
     ))
   }
+
+  pub(crate) fn check_mp4_codec(path: &Path) -> Result<(), Error> {
+    let f = File::open(path)?;
+    let size = f.metadata()?.len();
+    let reader = BufReader::new(f);
+
+    let mp4 = Mp4Reader::read_header(reader, size)?;
+
+    for track in mp4.tracks().values() {
+      if let TrackType::Video = track.track_type()? {
+        if let MediaType::H264 = track.media_type()? {
+          return Ok(());
+        } else {
+          return Err(anyhow!("h264 format supported, only"));
+        }
```

Comment:
I have no idea what can go into an MP4, but let's check all tracks, and also put the unsupported media type in the error message:

```
      if let TrackType::Video = track.track_type()? {
        let media_type = track.media_type()?;
        if let MediaType::H264 != media_type {
          return Err(anyhow!("Unsupported video codec, only H.264 is supported in MP4s: {media_type}"));
        }
```

Reply:
I just realized, the code I had before had slightly different logic than when I explained what I wanted... I'm gonna redo some of that logic also.